src/sentry/snuba/metrics/extraction.py

```diff
@@ -567,18 +567,12 @@ def combine(cls, *supported_by):
 
 
 def should_use_on_demand_metrics(
-    dataset: Optional[Union[str, Dataset]],
     aggregate: str,
     query: Optional[str],
     groupbys: Optional[Sequence[str]] = None,
 ) -> bool:
     """On-demand metrics are used if the aggregate and query are supported by on-demand metrics but not standard"""
     groupbys = groupbys or []
-    supported_datasets = [Dataset.PerformanceMetrics]
-    supported_datasets.append(Dataset.Transactions)
-
-    if not dataset or Dataset(dataset) not in supported_datasets:
-        return False
```

Comment:
Let me know if I got this right.

Reply:
so we need to keep the dataset check, but simplify it to:
```python
if not dataset or Dataset(dataset) != Dataset.PerformanceMetrics:
	return False
```

Reply:
I've adjusted it.